packages/web3-eth-contract/src/contract.ts

```diff
@@ -1050,7 +1051,9 @@ export class Contract<Abi extends ContractAbi>
 				{ requestManager: this.requestManager },
 			);
 
-			await this.subscriptionManager?.addSubscription(sub);
+			this.subscriptionManager?.addSubscription(sub).catch(() => {
+				throw new SubscriptionError('Failed to subscribe.');
+			});
```

Comment:
I would suggest to create an `error` event and `emit` the error on `sub` object in this particular case.